test/e2e/scale_test.go

```diff
@@ -0,0 +1,161 @@
+// +build e2e
+
+/*
+Copyright 2018 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"testing"
+	"time"
+
+	pkgTest "github.com/knative/pkg/test"
+	"github.com/knative/pkg/test/logging"
+	"github.com/knative/serving/pkg/apis/serving/v1alpha1"
+	serviceresourcenames "github.com/knative/serving/pkg/reconciler/v1alpha1/service/resources/names"
+	"github.com/knative/serving/test"
+	"golang.org/x/sync/errgroup"
+)
+
+func testScaleToWithin(t *testing.T, logger *logging.BaseLogger, scale int, duration time.Duration) {
+	clients := Setup(t)
+
+	var imagePath = test.ImagePath("helloworld")
+
+	deployGrp, _ := errgroup.WithContext(context.TODO())
```

Comment:
Probably `context.Background()` unless you expect we won't need this in the future.

Reply:
Done.